services/sidecar/src/simcore_service_sidecar/log_parser.py

```diff
@@ -78,8 +79,14 @@ async def monitor_logs_task(
 async def _monitor_docker_container(
     container: DockerContainer, log_cb: Awaitable[Callable[[LogType, str], None]]
 ) -> None:
-    async for line in container.log(stdout=True, stderr=True, follow=True):
-        log_type, parsed_line = await parse_line(line)
+    try:
+        async for line in container.log(stdout=True, stderr=True, follow=True):
+            log_type, parsed_line = await parse_line(line)
+    except aiodocker.exceptions.DockerError as e:
+        log_type, parsed_line = await parse_line(
+            f"Could not recover logs because: {str(e)}"
```

Comment:
just wondering, import aiodocker vs from aiodocker.exceptions import DockerError ? what are the implications? is there any typical python directive?

Reply:
there is no reason, but this way you know exactly from which module the error comes. It's more readable because sometimes you may have a `GenericError` and with this form `except module.submodule.GenericError...` is easier to read.